src/testing/TestFields.tsx

```diff
@@ -0,0 +1,151 @@
+                      form.setFieldValue(
+                        key,
+                        typeof option.const === 'boolean' ? target.value === 'true' : target.value
+                      )
+                    }
+                  />
+                  {option.title}
+                </Label>
+              ))}
+              {fieldError(key)}
+            </Box>
+          );
+        }
+
+        if (value.type === 'boolean') {
+          return (
+            <Box key={key}>
+              <Label>
+                <Checkbox {...form.getFieldProps(key)} />
+                <Box>
+                  {value.title} *
+                  {value.description ? (
+                    <Text as="small" sx={{ display: 'block' }}>
+                      {value.description}
+                    </Text>
+                  ) : null}
+                </Box>
+              </Label>
+              <Text>{fieldError(key)}</Text>
+            </Box>
+          );
+        }
+
+        return (
+          <Box key={key}>
+            <Label htmlFor={`test-${key}`}>{value.title} *</Label>
+            <Input
+              type={value.type === 'number' ? 'number' : 'text'}
+              id={`test-${key}`}
+              {...form.getFieldProps(key)}
+            />
+            {fieldError(key)}
+          </Box>
+        );
+      })}
+
+      <Box>
+        <Label htmlFor="test-notes">
+          <Message>addTestForm.notes.label</Message>
+        </Label>
+        <Textarea id="test-notes" {...form.getFieldProps('notes')} sx={{ resize: 'vertical' }} />
+      </Box>
+    </>
+  );
+};
+
+export function getInitialValues(schema: ObjectSchema) {
+  const initialValueEntries = Object.entries(schema.properties).map(([key, value]) => [
+    key,
+    initialPropertyValue(value),
+  ]);
+  return { ...Object.fromEntries(initialValueEntries), notes: '' };
+}
+
+export function getValidationSchema(schema: ObjectSchema) {
+  const requiredFields = schema.required || [];
+
+  const entries = Object.entries(schema.properties).map(([key, value]) => {
+    const schema = validationSchema(value, requiredFields.includes(key));
+    return [key, schema];
+  });
+  return Object.fromEntries(entries);
+}
+
+function validationSchema({ type, oneOf }: FieldSchema, required: boolean) {
+  let validation;
+  if (oneOf) {
+    validation = yup.mixed().oneOf(oneOf.map(({ const: value }) => value));
+  } else {
+    switch (type) {
+      case 'boolean':
+        validation = yup.boolean();
+        break;
+      case 'number':
+        validation = yup.number();
+        break;
+      case 'null':
+        validation = yup.string().nullable();
+        break;
+      case 'string': // fallthrough
+      default:
+        validation = yup.string();
+        break;
+    }
+  }
+
+  if (required) {
+    validation = validation.required();
+  }
+
+  return validation;
+}
+
+function initialPropertyValue({ type, oneOf }: FieldSchema) {
+  if (oneOf) {
+    // FIXME: What to do here?
```

Comment:
@Tankenstein @StevePole What do you think about this? Added this, as we don't want either of the options to be preselected on init.

Reply:
That's the correct logic, given the sensitivity of the data we don't want to preselect anything.

Reply:
✅ Removed the `FIXME`.